### DISCUSSION TITLE
`no field of name` error in SQL Lab

### DISCUSSION THREAD

**Discussion by leo-schick** [apache/superset#23213] [General]
In SQL Lab when I write a query which does not have a column name for a column I get the error `no field of name`.

E.g. use query
``` sql
SELECT 1
```

I have to write
``` sql
SELECT 1 AS col1
```

IMO I should not get this error in SQL Lab when querying data ad-hock. Superset should use an optimistic approach here e.g. setting a default column e.g. `col1`, `col2`, ...
I should only get this as error when I want to save the sql query as a data set.

Maybe a UI highlighting or a warning sign in the column name could be then implemented so that the user sees right away that the column name was generated dynamically in Superset and needs to be adjusted before saving it as Dataset.

Database backend: Azure Synapse
Superset Version: 1.5.3

**Comment by sfirke** [comment]
Related, I just filed this bug report - also an issue with unnamed columns in SQL Server: https://github.com/apache/superset/issues/23848

**Comment by sfirke** [comment]
Fixed in #38986
